trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestBuilder.java

```diff
@@ -205,8 +202,36 @@ public Optional<InputStream> inputStream()
         };
     }
 
-    private static String getRequestContentTypeFromHeader(MultiMap requestHeaders)
+    private static Optional<ContentType> getChunkingType(String valueToParse)
+    {
+        return switch (valueToParse.toLowerCase(Locale.ROOT).trim()) {
+            case "aws-chunked" -> Optional.of(ContentType.AWS_CHUNKED);
+            case "chunked" -> Optional.of(ContentType.W3C_CHUNKED);
+            default -> Optional.empty();
+        };
+    }
+
+    private static ContentType getRequestContentTypeFromHeaders(MultiMap requestHeaders)
     {
-        return requestHeaders.getFirst("content-encoding").or(() -> requestHeaders.getFirst("transfer-encoding")).orElse("");
+        List<String> requestContentEncoding = requestHeaders.get("content-encoding");
+        if (requestContentEncoding.isEmpty()) {
+            return getChunkingType(requestHeaders.getFirst("transfer-encoding").orElse("")).orElse(ContentType.STANDARD);
```

Comment:
Have we verified that only 1 `transfer-encoding` header is allowed?

Reply:
I have been experimenting with this, I will open another issue shortly to address that